### ISSUE TITLE
Signature Box Expansion to Sketch Box

### ISSUE THREAD

**Issue by RKnepper3** [Esri/Survey123Community#168]
I was wondering if there is talk of creating a new Appearance format for a sketch?  Basically the same as the signature format for image but a much larger box to make a sketch on?  My thought is if you are using this on a tablet and could have the sketch box be larger (preferably it would be a popup that takes up the entire screen) that lets users make a sketch.  It would be helpful for something like rough site drawing (where you do not need exact dimensions) with a stylus

**Comment by survey123** [comment]
Absolutely. This is part of the XLSForm spec already. It is achieved through the ['draw' appearance](http://survey123.esri.com/resources/XLSFormSupport/#appearance) in an 'image' question type. 
Thanks for bringing this to our attention.
Lets wait for votes on this one!

**Comment by RKnepper3** [comment]
Thanks, I didn't notice that in the doc before.  Look forward to it.

**Comment by RKnepper3** [comment]
Curious, was the plan to function exactly as the signature?  If you are doing something more involved like a site sketch you might make a mistake and want to not start completely, maybe erase or undo?

**Comment by RKnepper3** [comment]
Curious when we might hope to see this?  next year or possibly sooner?  I have a perfect project in mind once that is added but cant really do it without an expanded sketch area.

**Comment by ArianaToth** [comment]
I also need a sketch box.
https://geonet.esri.com/groups/survey123/blog/2015/08/23/signatures

**Comment by SonomaLandTrust** [comment]
Looking forward to the implementation of this draw type in addition to or as a follow up to implementing the signature type (an additional thanks for doing that).

**Comment by SonomaLandTrust** [comment]
Currently the work around for me is to use a third party sketch tool on the device, save the sketch as an image and use the image tool/widget to select the sketch instead of taking a photo.

**Comment by MKellyEsri** [comment]
+1 for this. Another enhancement could be to overlay sketches on a picture. So you could take a picture, and then add some notes on top of the image.

**Comment by RKnepper3** [comment]
Just wondered if there was a timeframe on this update?  Overlay on a picture (or map) would be nice but even just an expanded signature box to display for drawing would be a good addition

**Comment by survey123** [comment]
Hi Ronald. We are aiming at May the 4th. Updated Milestone.

**Comment by colgateg** [comment]
...sketch on a photo sounds super useful.. 

+1

**Comment by MariusBurger** [comment]
The ability to have a larger (than signature) sketch function will be great. Require such a function for drawing accident scenes

**Comment by pinpointmapping** [comment]
+1 for this.  Handy for documenting Water Breaks too

**Comment by akilloran5** [comment]
Just wondering if there has been an update on the sketch tool? I have a customer who would be very interested in this tool for energy surveys on houses. They currently free hand draw the house and windows and write in measurements but it would be great if Survey123 could do this and then do the calculations based off measurements to find the area of the rooms.

**Comment by brycebauer** [comment]
I am wondering if there is an update on this sketch box? I am currently working on a survey that would like to have this feature to sketch any issues on buildings.

**Comment by sirws** [comment]
+1

**Comment by brendanpauls** [comment]
+1 to this idea.  It would be super useful.  Sounds like it may still be addressed in 2016 as per Ismael's comments on GeoNet.  Hope so!

**Comment by AMHOWL** [comment]
+1
this would be fantastic!  Very useful.

**Comment by travisbutcher** [comment]
👍

**Comment by jrugys** [comment]
+1 
This would be very useful for a wastewater utility customer who would like to include simple sketches in their tap request forms.

**Comment by Nick-t-go** [comment]
+1

**Comment by jbhyperlite39** [comment]
+1

**Comment by talllguy** [comment]
@survey123 any updates on a timeline for this enhancement? Kudos on the hard work so far!

**Comment by tedrick** [comment]
@talllguy - Nothing concrete at this point.  There has been some work towards this for another project that we hope to take advantage of in a future release.

**Comment by talllguy** [comment]
Keep innovating! thanks for the info.

**Comment by CamillaIsaksson** [comment]
@survey123 Agree with previous..It would be a very nice to have the ability to draw on top of an image. Exactly as the signatur works today in Survey123 but with the possibility to add a background image. Good for profile drawings..

**Comment by kaptinnemo** [comment]
Any updates on the sketch box expansion?

**Comment by tedrick** [comment]
We've been doing a fair amount of work on the underlying component, but are encountering a few issues at the Qt (programming) level that need to be resolved before implementing.

**Comment by kaptinnemo** [comment]
I have been using the draw feature, but I am running into problems where after I draw a few lines, the older parts of the drawing begin to be erased. Has anyone else experienced this problem?

**Comment by JohnHasthorpe** [comment]
@kaptinnemo Please can you share the device and operating system you are using?
